src/controller/CHIPDeviceController.cpp

```diff
@@ -1202,6 +1218,206 @@ void DeviceCommissioner::OnSessionEstablished()
+                                        ->GetCryptoContext()
+                                        .GetAttestationChallenge();
+
+    AttestationVerificationResult result = dac_verifier->VerifyAttestationInformation(
+        attestationElements, attestationChallenge, signature, device->GetPAI(), device->GetDAC(), device->GetAttestationNonce());
+    if (result != AttestationVerificationResult::kSuccess)
+    {
+        if (result == AttestationVerificationResult::kNotImplemented)
+        {
+            ChipLogError(Controller,
+                         "Failed in verifying 'Attestation Information' command received from the device due to default "
+                         "DeviceAttestationVerifier Class not being overriden by a real implementation.");
+            return CHIP_ERROR_NOT_IMPLEMENTED;
+        }
+        else
+        {
+            ChipLogError(Controller,
+                         "Failed in verifying 'Attestation Information' command received from the device: err %hu. Look at "
+                         "AttestationVerificationResult enum to understand the errors",
+                         static_cast<uint16_t>(result));
+            // Go look at AttestationVerificationResult enum in src/credentials/DeviceAttestationVerifier.h to understand the
+            // errors.
+            return CHIP_ERROR_INTERNAL;
+        }
+    }
+
+    ChipLogProgress(Controller, "Successfully validated 'Attestation Information' command received from the device.");
+
+    // TODO: Validate Certification Declaration
+    // TODO: Validate Firmware Information
+
+    return CHIP_NO_ERROR;
+}
+
+void DeviceCommissioner::HandleAttestationResult(CHIP_ERROR err)
+{
+    // Here we assume the Attestation Information validation always succeeds.
+    // Spec mandates that commissioning shall continue despite attestation fails (in some cases).
+    // TODO: Handle failure scenarios where commissioning may progress regardless.
+    if (err == CHIP_NO_ERROR)
+    {
+        VerifyOrReturn(mState == State::Initialized);
+        VerifyOrReturn(mDeviceBeingPaired < kNumMaxActiveDevices);
+
+        Device * device = &mActiveDevices[mDeviceBeingPaired];
+
+        ChipLogProgress(Controller, "Sending 'CSR request' command to the device.");
+        CHIP_ERROR error = SendOperationalCertificateSigningRequestCommand(device);
+        if (error != CHIP_NO_ERROR)
+        {
+            ChipLogError(Controller, "Failed in sending 'CSR request' command to the device: err %s", ErrorStr(error));
+            OnSessionEstablishmentError(error);
+            return;
+        }
+    }
+    else
+    {
+        // Handle error, and notify session failure to the commissioner application.
+        ChipLogError(Controller, "Failed to validate the Attestation Information");
+        // TODO: Map error status to correct error code
+        OnSessionEstablishmentError(CHIP_ERROR_INTERNAL);
```

Comment:
Please remove this for now to avoid a breaking change

Reply:
done